source/internal/_curryN.js

```diff
@@ -34,7 +34,14 @@ export default function _curryN(length, received, fn) {
       }
```

Comment:
I think we can track this without the additional loop.

We can add a new var up top (say line 20), `var hasPlaceholderBeforeFinalArg = false;`

And then here after updating combined we can add an else if

```js
if (!_isPlaceholder(result)) {
  left -= 1;
} else if (combinedIdx < length) {
  hasPlaceholderBeforeFinalArg = true;
}
```
and then at the end, similar to your current check: `return (left <= 0 && !hasPlaceholderBeforeFinalArg) ... `